thrall/app/lib/ThrallMessageConsumer.scala

```diff
@@ -39,21 +39,15 @@ object ThrallMessageConsumer extends MessageConsumer(
   def updateImageUserMetadata(metadata: JsValue): Future[UpdateResponse] =
     withImageId(metadata)(id => ElasticSearch.applyImageMetadataOverride(id, metadata \ "data"))
 
-  def deleteImage(image: JsValue): Future[EsResponse] =
+  // The Unit, Unit is due to the two sied effects
+  def deleteImage(image: JsValue): Future[Either[ImageDeletedResponse, ImageNotDeletableResponse]] =
     withImageId(image) { id =>
-      // if we cannot delete the image as it's "protected", succeed and delete
-      // the message anyway.
-      ElasticSearch.deleteImage(id).map {
-        case r: DeleteByQueryResponse => {
+      ElasticSearch.deleteImage(id).map { deleteResponse =>
+        deleteResponse.left.map { imageDeletedResponse =>
```

Comment:
The reason being that we definitely care that the image was allowed to be deleted i.e. an image with that `id` and `deletableFilter` exists and was deleted.

We were using a future before, but that could fail as the request was unsuccessful i.e. Elastic broke, or because we decided it would fail because it wasn't deltable.

We also use to care if the image was not deleted so we could log it - but I moved the logging into eh ES client as that's where the rest of the logic if. I suppose this could be a `Future[Option[ImageDeletedResponse]]` as we no longer care about the failure of an image to be deleted.